src/Analyzer.JsonRuleEngine.UnitTests/InOperatorTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.UnitTests
+{
+    [TestClass]
+    public class InOperatorTests
+    {
+        [DataTestMethod]
+        [DataRow(true, 4, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Integer is in")]
+        [DataRow(false, 8, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Integer is not in")]
+        [DataRow(true, 3.5, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Float is in")]
+        [DataRow(false, 9.5, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Float is not in")]
+        [DataRow(true, "aValue", new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "String is in")]
+        [DataRow(false, "someOtherValue", new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "String is not in")]
+        [DataRow(true, false, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Boolean is in")]
+        [DataRow(false, true, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Boolean is not in")]
+        [DataRow(true, null, new object[] { 4, 3.5, "aValue", false, null }, DisplayName = "Null is in")]
+        [DataRow(false, null, new object[] { 4, 3.5, "aValue", false }, DisplayName = "Null is not in")]
+        [DataRow(false, 7, new object[] { }, DisplayName = "Integer is not in empty array")]
+        [DataRow(false, 7.7, new object[] { }, DisplayName = "Float is not in empty array")]
+        [DataRow(false, "aValue", new object[] { }, DisplayName = "String is not in empty array")]
+        [DataRow(false, true, new object[] { }, DisplayName = "Boolean is not in empty array")]
+        [DataRow(false, null, new object[] { }, DisplayName = "Null is not in empty array")]
+        public void EvaluateExpression(bool evaluationResult, object desiredValue, object arrayOfValues)
```

Comment:
I see how I could do that for the null test, something like: EvaluateExpression_NullAsTokenToEvaluate_ReturnsFalse, right? But for the other test the outcome and the case being tested changes in each DataRow. It's okay if I leave that one the way it is?

Reply:
You can do it in a generic way. Something like EvaluateExpression_ValidDataType_ReturnsExpectedEvaluationResult

Reply:
The Equals operator tests have good examples, like `EvaluateExpression_PropertyIsEqual_EqualsExpressionIsTrue_NotEqualsExpressionIsFalse`.  It might be good to separate this test into two - one where the value is in the array, one where it isn't.